rfcs/0076-refined-s3-source.md

```diff
@@ -0,0 +1,88 @@
+---
+feature: Refined S3 Source
+authors:
+  - "TabVersion"
+start_date: "2023/09/08"
+---
+
+# RFC: Refined S3 Source
+
+## Background and Motivation
+
+There has been two RFCs ([RFC: Refine S3 Source: List Objects in A Lazy Way](https://github.com/risingwavelabs/rfcs/pull/74) 
+and [RFC: S3 Source with SQS](https://github.com/risingwavelabs/rfcs/pull/22)) talking about S3 source but neither of them 
+fits the need of RisingWave. This RFC is to combine the pros of both RFCs and make S3 source elegant.
+
+The main problems are:
+
+* There can be millions of files in s3, but we cannot store all items in Meta's memory or ETCD.
+* Pushing down the file assignment from Meta to Compute Node can be burdensome, and it makes barrier stuck if it carries 
+  too many files.
+
+# Solution
+
+The intuitive solution is to use state store to store the file list, and it is kind of a good idea.
+But it is not easy for Meta to access Hummock, the storage engine, and even if it could write the file list down to state
+store, it is even harder to align the data with barrier, which means it can be dangerous for source executors to use the list.
+
+## List and Fetch
+
+So I am proposing a new solution, which is to use two separate executors to complete the job.
+Let's say they are `list` and `fetch`, aligning the name in [RFC: S3 Source with SQS](https://github.com/risingwavelabs/rfcs/pull/22), 
+and they do the same thing.
+
+* `List` is unique in one source plan. It is responsible for keep listing all files and push the filenames to the downstream.
```

Comment:
Which layer does `List` reside in?
- A new kind of connector (split) in the source executor, which persists its state in the offset.
- A new executor.